lib/modules/manager/gradle/extract/catalog.ts

```diff
@@ -39,26 +49,83 @@ function extractVersion({
   versionStartIndex,
   versionSubContent,
 }: {
-  version: string | VersionPointer;
-  versions: Record<string, string>;
+  version: GradleVersionCatalogVersion;
+  versions: Record<string, GradleVersionPointerTarget>;
   depStartIndex: number;
   depSubContent: string;
   depName: string;
   versionStartIndex: number;
   versionSubContent: string;
+}): VersionExtract {
+  if (isVersionPointer(version)) {
+    // everything else is ignored
+    return extractLiteralVersion({
+      version: versions[version.ref],
+      depStartIndex: versionStartIndex,
+      depSubContent: versionSubContent,
+      sectionKey: version.ref,
+    });
+  } else {
+    return extractLiteralVersion({
+      version: version,
+      depStartIndex,
+      depSubContent,
+      sectionKey: depName,
+    });
+  }
+}
+
+function extractLiteralVersion({
+  version,
+  depStartIndex,
+  depSubContent,
+  sectionKey,
+}: {
+  version: GradleVersionPointerTarget;
+  depStartIndex: number;
+  depSubContent: string;
+  sectionKey: string;
 }): VersionExtract {
   if (!version) {
-    return {};
+    return { skipReason: 'no-version' };
+  } else if (typeof version === 'string') {
+    const fileReplacePosition =
+      depStartIndex + findIndexAfter(depSubContent, sectionKey, version);
+    return { currentValue: version, fileReplacePosition };
+  } else if (typeof version === 'object') {
```

Comment:
```suggestion
  } else if (is.plainObject<RichVersion>(version)) {
```

Reply:
This suggestion seems incorrect -- this would assert? that the members of `version` have the type `RichVersion`, when instead `version` itself is a `RichVersion`. I've used `is.object` instead -- let me know if there's a more appropriate option.

Reply:
Nope, read the `is` docs for difference about `object` and `plainObject`.